### PR TITLE
Phac endpoint redux

### PR DESCRIPTION
**Pull request by ashtonmeuser**

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -85,27 +85,43 @@ app.post(`${apiBaseUrl}/form`,
...
+    })));
+
+    const results = { successful: {}, duplicates: [], errors: [] };
+    try {
+      await phacCollection.insertMany(phacItems, { ordered: false });
+      results.successful = phacItems.reduce((a, v) => ({ ...a, [v.covid_id]: v.id }), {});
+    } catch (error) {
+      try {
+        const { writeErrors } = error.result.result;
+        results.duplicates = writeErrors
```
Comment (fwkendall) [inline server/server.js:110]: Am I missing something? The logic for identifying duplicates vs. errors looks identical to me....

File: server/server.js [hunk 2]
```diff
@@ -85,27 +85,43 @@ app.post(`${apiBaseUrl}/form`,
...
+          .filter((e) => e.err.code === 11000)
+          .map((e) => e.err.op.covid_id);
+        results.errors = writeErrors
+          .filter((e) => e.err.code !== 11000)
+          .map((e) => e.err.op.covid_id);
+        results.successful = phacItems.filter((i) => (
+          !results.duplicates.includes(i.covid_id) && !results.duplicates.includes(i.covid_id)
```
Comment (fwkendall) [inline server/server.js:117]: Is this the same condition twice?

Comment (mauriciofreshw) [inline server/server.js:117]: Did you mean `!results.duplicates.includes(**i.id**) && !results.duplicates.includes(i.covid_id)` ?

File: server/server.js (hunk 1 above)
Comment (ashtonmeuser) [reply server/server.js:110]: Ah lol the code returned by the MongoDB driver is `11000` for duplicates. So it's `===` for dupes and `!==` for everything else (which we consider errors).

File: server/server.js (hunk 2 above)
Comment (ashtonmeuser) [reply server/server.js:117]: `=== 11000` vs `!== 11000`

Comment (ashtonmeuser) [reply server/server.js:117]: 👆Ignore that. I see what you're saying. Line 117.

Comment (ashtonmeuser) [reply server/server.js:117]: Whoops. Meant not in dupes and not in errors. Will fix.

File: server/server.js (hunk 1 above)
Comment (fwkendall) [reply server/server.js:110]: Ahhh that makes sense, thanks

**Comment by fwkendall** [comment]
Looks good 👍

**Pull request merged**